### ISSUE TITLE
New YT app version reporting thread

### ISSUE THREAD

**Issue by th3an7** [pylerSM/YouTubeBackgroundPlayback#368]
# YT app just got updated and module's not working? Report it below with new version numbers and wait for an update :)

**Comment by thebestnom** [comment]
I have no idea why apkmirror thinks 14.09.52 is beta, but it looks like that play store update to it without being a beta tester

**Comment by th3an7** [comment]
In that case I'll add that version later...
Will edit this when it's done so refresh this in 3-4h maybe :P

**Comment by th3an7** [comment]
@thebestnom Sorry it took me so long... but it should work again :D

**Comment by thebestnom** [comment]
14.09.53 just got out...
Youtube got updated twice in a week, thats not normal at all, I wonder what they are trying to do

**Comment by th3an7** [comment]
Probably just quick bug fixes... Will add it later tonight

**Comment by th3an7** [comment]
@thebestnom fixed, should work again - it was just a quick bug fix probably since classes are the same and it was only minor version bump... It happens from time to time

**Comment by thebestnom** [comment]
Works as always, thanks! But the obfuscation happens every build, So class name should change regardless, am I wrong?

**Comment by th3an7** [comment]
It should but when the version is bumped by .01 (like in this case), they are the same - I don't know if they are releasing the same app with changed version number or if they're changing something but it happens almost every time :P

**Comment by thebestnom** [comment]
14.10.53 is not beta...

**Comment by th3an7** [comment]
@thebestnom Updated... I couldn't understand that to this day...
AppManifest doesn't show if it's beta or not so I must guess if it is or not - good to have you here :D

**Comment by thebestnom** [comment]
well, after you are being super fast for at least a month (I think) youtube have  got you :-P
version 14.17.53 is out

**Comment by th3an7** [comment]
@thebestnom I knew it will come sooner than later - I think I saw it in the beta so I've been waiting until it comes to stable - thanks for the info :D 

As always - it's updated and should work again

**Comment by thebestnom** [comment]
Works as always! Thanks again
Btw, just updated to Android PIE, and using edxposed and everything works great!

**Comment by th3an7** [comment]
Yeah... I've tried it myself but it was crashing a lot so I had to
uninstall edxposed and stay with Vanced...
Glad it works now :D

**Comment by abhijeetramgir** [comment]
Hey! I have youtube version 14.20.54 on Android 8.1, YBP is not working.
Any help? 
Thanks!

**Comment by th3an7** [comment]
@abhijeetramgir it's not yet supported - I'll add it today, sit tight (will edit this when it's done)

**Comment by remewer** [comment]
Doesn't work on 14.23.50

**Comment by ohyeah521** [comment]
> Doesn't work on 14.23.50

try this ,may be work. 
https://repo.xposed.info/module/wovow.youtubehelper

**Comment by th3an7** [comment]
@remewer should work again

Thanks to @ohyeah521 for the hooks :D

**Comment by thebestnom** [comment]
YouTube released a new version 14.28.54

**Comment by th3an7** [comment]
> YouTube released a new version 14.28.54

I saw it - will add it in a second...
I've also seen 14.29.53 and .55 coming next that's why I've been waiting with adding it - but if the 28.54 is in Play Store already, I'll add it now...

I'll edit this when it's done (probably max. 1h)

**Comment by thebestnom** [comment]
Oh, wierd that they are doing so many versions in parallel, but thanks as always!

**Comment by th3an7** [comment]
> Oh, wierd that they are doing so many versions in parallel, but thanks as always!

It happens sometimes - probably they're just quickly patching bugs and preparing new version or testing some server-side changes...

Anyway - it should work again for 14.28.54 as well as for 14.28.55, 14.29.53 and 14.29.55 - just force close app or reboot your device :D

**Comment by remewer** [comment]
Doesn't work on 14.59.26 yet

**Comment by th3an7** [comment]
> Doesn't work on 14.59.26 yet

I believe you meant 14.29.56 - can you check if it works now? Haven't edited anything, just copy-pasted old ones with changed version number - guessing classes haven't changed - if it still doesn't work I'll edit them in a second :)

**Comment by remewer** [comment]
It does work now. Thanks a lot

**Comment by remewer** [comment]
Doesn't work on 14.31.50

**Comment by th3an7** [comment]
Sure... Will add it later when I find a while if that's ok :P

**Comment by th3an7** [comment]
Sorry for the delay... 14.31.50 has been added and module should be working again

**Comment by Kronopt** [comment]
Can you please add hooks for youtube version 14.31.51?

**Comment by remewer** [comment]
> Can you please add hooks for youtube version 14.31.51?

It works just fine for me. Same version

**Comment by th3an7** [comment]
> Can you please add hooks for youtube version 14.31.51?

Sorry for the delay... Got some personal problems I need to deal with...

Anyway - it should work again - I've added 14.31.51 and 14.31.52
Also thanks to @ohyeah521, hooks for 14.32.51 were added too

**Comment by remewer** [comment]
Please add support for 14.33.56. Thanks 😉

**Comment by th3an7** [comment]
> Please add support for 14.33.56. Thanks 😉

Added - force close YT app to make it work again

**Comment by thebestnom** [comment]
New version 14.37.54 is out

**Comment by th3an7** [comment]
> New version 14.37.54 is out

Seems like there is only .apk for 4.4+ - I'll wait until someone uploads package for 5.0+

**Comment by thebestnom** [comment]
Version 14.38.53 is out, but it looks again that apk mirror have only the 4.4+ version, shouldn't it be automatic?

**Comment by th3an7** [comment]
> Version 14.38.53 is out, but it looks again that apk mirror have only the 4.4+ version, shouldn't it be automatic?

Nah, people are uploading files to them, they are reviewing if everything is correct (original .apk without modified malicious code) and then it appears as available...

Anyway it should work again thanks to @ohyeah521

**Comment by remewer** [comment]
Please add 14.39.55 https://www.apkmirror.com/apk/google-inc/youtube/youtube-14-39-55-release/

**Comment by remewer** [comment]
Please add 14.41.52 https://www.apkmirror.com/apk/google-inc/youtube/youtube-14-41-52-release/

**Comment by thebestnom** [comment]
14.42.54 is out

**Comment by th3an7** [comment]
Updated - force close YT app

**Comment by wjj1928** [comment]
Youtube 14.44.52 is out, please help update! BTW, would you please help share the steps about how to upgrade to a new version? not sure if it's easy for me to upgrade myself! :-), but I wish to try!

**Comment by th3an7** [comment]
> Youtube 14.44.52 is out, please help update! BTW, would you please help share the steps about how to upgrade to a new version? not sure if it's easy for me to upgrade myself! :-), but I wish to try!

Not seeing .apk on apkmirror or anywhere else yet - can you please upload it somewhere?
Also - it would be better to stop auto updates from Play Store so you'll always be sure you're on latest working version.

**Comment by wjj1928** [comment]
> > Youtube 14.44.52 is out, please help update! BTW, would you please help share the steps about how to upgrade to a new version? not sure if it's easy for me to upgrade myself! :-), but I wish to try!
> 
> Not seeing .apk on apkmirror or anywhere else yet - can you please upload it somewhere?
> Also - it would be better to stop auto updates from Play Store so you'll always be sure you're on latest working version.
> 
> Also it looks like it's beta (?) - module will be updated for those only if someone asks... and since you did that, I'll add it for you - just need .apk file (and for which android version it's for - 5.0+ or 4.2+)

I have uploaded a copy to [here](https://github.com/wjj1928/sharefolder/blob/master/com.google.android.youtube-14.44.52.apk). Maybe it's region specific. I am in USA. Thanks for your quick response!

**Comment by th3an7** [comment]
Updated to 14.44.52 for android 5.0+ (assuming it was for that) - should work again, just force close YT app from settings...

As for tutorial - I really should write something - I'll try to add something when I find a while...

**Comment by wjj1928** [comment]
Thanks, 14.44.52 works now but  got below exception from logs:

11-04 14:50:40.611 24547 24547 I YTBackgroundPlayback: Done applying the hooks. [vc:1444523000]
11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: java.lang.NoSuchFieldError: java.lang.Boolean#b
11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: 	at de.robv.android.xposed.XposedHelpers.findField(Unknown Source:71)
11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: 	at de.robv.android.xposed.XposedHelpers.setBooleanField(Unknown Source:4)
11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: 	at com.pyler.youtubebackgroundplayback.YouTubeBackgroundPlayback$3.beforeHookedMethod(YouTubeBackgroundPlayback.java:194)
11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: 	at de.robv.android.xposed.XC_MethodHook.callBeforeHookedMethod(Unknown Source:0)
11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: 	at EdHooker1.hook(Unknown Source:84)

**Comment by th3an7** [comment]
> Thanks, 14.44.52 works now but got below exception from logs:
> 
> 11-04 14:50:40.611 24547 24547 I YTBackgroundPlayback: Done applying the hooks. [vc:1444523000]
> 11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: java.lang.NoSuchFieldError: java.lang.Boolean#b
> 11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: at de.robv.android.xposed.XposedHelpers.findField(Unknown Source:71)
> 11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: at de.robv.android.xposed.XposedHelpers.setBooleanField(Unknown Source:4)
> 11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: at com.pyler.youtubebackgroundplayback.YouTubeBackgroundPlayback$3.beforeHookedMethod(YouTubeBackgroundPlayback.java:194)
> 11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: at de.robv.android.xposed.XC_MethodHook.callBeforeHookedMethod(Unknown Source:0)
> 11-04 14:50:48.058 24547 24547 E EdXposed-Bridge: at EdHooker1.hook(Unknown Source:84)

I've compared all classes from the newest version with some older one and they are all the same - this shouldn't happen... and I don't know what causes this
Did that log appear for older version or only for this one - and if module still works, you could ignore that - unless this error is logged constantly when YT app is force closed/phone rebooted - then there could be other field to hook but everything seems fine for me...

**Comment by woeijiunn88** [comment]
14.44.55 is out.😁

**Comment by th3an7** [comment]
> 14.44.55 is out.😁

Added - sorry for the delay... I now need to search different sites for .apk files as apkmirror only shows version for Android 4.2+ and it's missing latest 2 :P

**Comment by woeijiunn88** [comment]
damn youtube updated again to 14.45.52
donno this will helpful for you or not: https://apk.support/app/com.google.android.youtube#latest_version

**Comment by th3an7** [comment]
> damn youtube updated again to 14.45.52
> donno this will helpful for you or not: https://apk.support/app/com.google.android.youtube#latest_version

Updated
I really do recommend disabling automatic updates in Play Store tho - app doesn't change that much in terms of functionality or design and yet it gets updated every 3 days breaking YTBP module...

**Comment by woeijiunn88** [comment]
> I really do recommend disabling automatic updates in Play Store tho - app doesn't change that much in terms of functionality or design and yet it gets updated every 3 days breaking YTBP module...

Yea i think i nid to do detach from play store on my youtube app ltr

**Comment by thebestnom** [comment]
Welp, another week another new version...
Version 14.46.52 is out

**Comment by th3an7** [comment]
> Welp, another week another new version...
> Version 14.46.52 is out

Added... although I am not sure if it works cause there is no .apk for .52 - I've updated it looking at 14.46.53 and I assumed they are both the same...

Anyway - as I said before - you really should disable automatic updates of YT app in Play Store so YTBP won't break everytime you'll update... one update per month should be enough as even old versions are getting new design and features via server side updates...

**Comment by thebestnom** [comment]
Yeah, Im mostly doing so, but I accidentally clicked on the update all button 😅 thanks as always!

**Comment by preethamrn** [comment]
Any chance you'll release a version with the YTM background play in the same package from #381. I'm still on v2.0.5 from [#357 (comment)](https://github.com/pylerSM/YouTubeBackgroundPlayback/issues/357#issuecomment-379087066).

**Comment by th3an7** [comment]
Let me see...
I won't promise anything but if I still have Android Studio set up, I may have something... I'll either edit this message, make new Issue or edit the old one with something new :)

But as I said - don't make your hopes too high :P

**Comment by lrdmic** [comment]
14.46.52 version is working with new module.

**Comment by DanGLES3** [comment]
14.50.53 just released

**Comment by th3an7** [comment]
> 14.50.53 just released

Updated - also added 14.50.52 - has the same classes as .53 so both should work again

**Comment by thebestnom** [comment]
hey, version 14.51.51 is out (for a while) I just recently moved phones, so if you don't mind it would be really helpful if you could add it!

**Comment by th3an7** [comment]
> hey, version 14.51.51 is out (for a while) I just recently moved phones, so if you don't mind it would be really helpful if you could add it!

Added - hopefully works again

**Comment by thebestnom** [comment]
works great even on android 10 (still trying to understand how the hell popup with gestures work)! Thank you so much!

**Comment by hiishamz** [comment]
Just noticed I'm running version 15.03.56

**Comment by th3an7** [comment]
> Just noticed I'm running version 15.03.56

Updated

**Comment by hemantmeena1990** [comment]
is module update itself automatically from git?

**Comment by th3an7** [comment]
Yes - every time you open YT app, it checks and downloads hooks

That's why I always say to force close YT app - to force download updated
hooks file

**Comment by bilalt2891** [comment]
how do i download this mod as the xposed repo seems to be down?

**Comment by th3an7** [comment]
> how do i download this mod as the xposed repo seems to be down?

Check here -> https://github.com/pylerSM/YouTubeBackgroundPlayback/issues/365#issuecomment-557856132

**Comment by dcoolArne** [comment]
The background playback works, but i don't have any media controls in the latest version

**Comment by th3an7** [comment]
> The background playback works, but i don't have any media controls in the latest version

Hmm... Module only enables it - it shouldn't block/hide it - check if you have YT notifications enabled in settings (Android OS settings ofc)

Classes are 100% correct - if everything was working before, then it may be app bug and not module fault...

**Comment by dcoolArne** [comment]
Yeah I have all notifications enabled, it happend after the last update. If I reset the app I get a notification about YouTube premium settings but there is nothing about media controls.

**Comment by th3an7** [comment]
Can you downgrade to one version before? It may actually be a bug in YT app
- I didn't change anything else besides standard classes update...

**Comment by dcoolArne** [comment]
Can't really do that right now, but will try tomorrow

**Comment by dcoolArne** [comment]
I deactivated the app and then updated it trough the play store again, now everything is back to normal

**Comment by infoflo7** [comment]
Latest YouTube update disables YouTube background playback.

**Comment by th3an7** [comment]
> Latest YouTube update disables YouTube background playback.

Updated for 15.12.33 - hopefully it's the latest one, force close and it should work again

**Comment by infoflo7** [comment]
Thanks for the fix on the last YouTube version. The newest one (which came out today or yesterday) 15.13.33 also disables background playback. Can you fix it?

**Comment by th3an7** [comment]
> Thanks for the fix on the last YouTube version. The newest one (which came out today or yesterday) 15.13.33 also disables background playback. Can you fix it?

Added - force close as always to make it work again

**Comment by pasirin** [comment]
The module stop working with YouTube version 15.14.33

**Comment by th3an7** [comment]
> The module stop working with YouTube version 15.14.33

Sorry for delay - updated, force close YT app 😄

**Comment by chrisk44** [comment]
The module stopped working with YT version 15.15.38.

**Comment by th3an7** [comment]
> The module stopped working with YT version 15.15.38.

Updated, as always - just force close YT app

**Comment by chrisk44** [comment]
Wow, that was fast. Thanks, it works!!

**Comment by thebestnom** [comment]
I accidently updated to 15.17.38, do you mind updating?

**Comment by th3an7** [comment]
> I accidently updated to 15.17.38, do you mind updating?

Not at all... seems like hooks were the same - should work again

**Comment by thebestnom** [comment]
That's weird... It's still doesn't work (after force close or phone restart)

**Comment by th3an7** [comment]
> That's weird... It's still doesn't work (after force close or phone restart)

Oops... seems like I've fucked up and used wrong .apk - now it should work... I am 99.9% sure... sorry :(

**Comment by thebestnom** [comment]
> > That's weird... It's still doesn't work (after force close or phone restart)
> 
> Oops... seems like I've fucked up and used wrong .apk - now it should work... I am 99.9% sure... sorry :(

Thank you as always! don't worry, you're doing this in your spare time, which is a great help anyway!

**Comment by someone5678** [comment]
Version 15.18.36 is out and module not working.

**Comment by th3an7** [comment]
> Version 15.18.36 is out and module not working.

Updated - should be back again

**Comment by someone5678** [comment]
> > Version 15.18.36 is out and module not working.
> 
> Updated - should be back again

Thx!!

**Comment by someone5678** [comment]
15.19.34 is out and module doesn't work again...

**Comment by th3an7** [comment]
> 15.19.34 is out and module doesn't work again...

Updated... there was some changes in latest class so I am not 100% sure if it's working... hopefully it is :P

**Comment by someone5678** [comment]
> > 15.19.34 is out and module doesn't work again...
> 
> Updated... there was some changes in latest class so I am not 100% sure if it's working... hopefully it is :P

Working perfectly!

**Comment by someone5678** [comment]
15.20.33 is out and stopped working 🥵
They changes too frequently..

**Comment by th3an7** [comment]
> 15.20.33 is out and stopped working 🥵
> They changes too frequently..

Can you upload apk? It's marked as beta on apkmirror but I can't download anything from there as it's bundle (`.apks`, installable through application... cannot be extracted...)

**Comment by remewer** [comment]
> > 15.20.33 is out and stopped working hot_face
> > They changes too frequently..
> 
> Can you upload apk? It's marked as beta on apkmirror but I can't download anything from there as it's bundle (`.apks`, installable through application... cannot be extracted...)

Actually, you can extract .apks file to obtain the base and split APKs.

**Comment by th3an7** [comment]
> > > 15.20.33 is out and stopped working hot_face
> > > They changes too frequently..
> > 
> > 
> > Can you upload apk? It's marked as beta on apkmirror but I can't download anything from there as it's bundle (`.apks`, installable through application... cannot be extracted...)
> 
> Actually, you can extract .apks file to obtain the base and split APKs.

Oops... I meant `.apkm` (shitty apkmirror file extension)... I found a tool for decrypting tho which converts `.apkm` to `.apks` - I'll update hooks in a while (expect it in the next. 20-30 minutes max)

I'll update this comment when it's done

**Comment by someone5678** [comment]
> > > > 15.20.33 is out and stopped working hot_face
> > > > They changes too frequently..
> > > 
> > > 
> > > Can you upload apk? It's marked as beta on apkmirror but I can't download anything from there as it's bundle (`.apks`, installable through application... cannot be extracted...)
> > 
> > 
> > Actually, you can extract .apks file to obtain the base and split APKs.
> 
> Oops... I meant `.apkm` (shitty apkmirror file extension)... I found a tool for decrypting tho which converts `.apkm` to `.apks` - I'll update hooks in a while (expect it in the next. 20-30 minutes max)
> 
> EDIT: Updated, should be back again...

dang it I missed to check the noti... sorry..
It working good! Thx!!!

**Comment by someone5678** [comment]
Version 15.22.33 is out!
And module stopped working..
Apkmirror doesn't updated yet so I extract it with SAI and upload to temp file uploads

https://tmpstorage.com/download/ab4hri7aav

**Comment by th3an7** [comment]
> Version 15.22.33 is out!
> And module stopped working..
> Apkmirror doesn't updated yet so I extract it with SAI and upload to temp file uploads
> 
> https://tmpstorage.com/download/ab4hri7aav

Seems like it is on apkmirror but it's marked as beta... weird...
Updated anyway 😄 

PS: I also like to check for new versions on [apkcombo.com](https://apkcombo.com/en-pl/apk-downloader/?q=com.google.android.youtube) but the problem is they're not including code version numbers (that I need to add to hooks cause of change in YT app code)... but at least .apk is there 😛

**Comment by someone5678** [comment]
https://apkcombo.com/en-pl/apk-downloader/?q=com.google.android.youtube

15.23.35 released!

**Comment by th3an7** [comment]
> https://apkcombo.com/en-pl/apk-downloader/?q=com.google.android.youtube
> 
> 15.23.35 released!

Updated

**Comment by someone5678** [comment]
https://ztp2.down-apk.com/download/50893311-e494-4e5f-a7ad-e72380a1f7cb?name=youtube_15.24.38.zip

15.24.38 is out now. But ApkCombo shows 15.24.37 as current version and 15.24.38 for other version 🤔

**Comment by th3an7** [comment]
> https://ztp2.down-apk.com/download/50893311-e494-4e5f-a7ad-e72380a1f7cb?name=youtube_15.24.38.zip
> 
> 15.24.38 is out now. But ApkCombo shows 15.24.37 as current version and 15.24.38 for other version 🤔

That cause "other versions" is taken from apkmirror - it probably doesn't matter as it seems like they did minor version bump and classes are the same in both .apk
I'll update it today - same as hooks for YTM

**Comment by th3an7** [comment]
Updated - had troubles with my PC so it took a little longer than usual... I think I've got it tho and it should work again for both YT and YTM
Classes taken from .37 but I added them as .38 so hopefully it works - if not, let me know and I do another update
(sending this as new message to send notifications 😅 )

**Comment by someone5678** [comment]
> Updated - had troubles with my PC so it took a little longer than usual... I think I've got it tho and it should work again for both YT and YTM
> Classes taken from .37 but I added them as .38 so hopefully it works - if not, let me know and I do another update
> (sending this as new message to send notifications  )

YT and YTM both working good!!!
I really really thanks to you 🥳

**Comment by someone5678** [comment]
https://apkcombo.com/en-pl/apk-downloader/?q=com.google.android.youtube

YT 15.26.34 is out!

**Comment by th3an7** [comment]
> https://apkcombo.com/en-pl/apk-downloader/?q=com.google.android.youtube
> 
> YT 15.26.34 is out!

Updated - YTM hooks too - both should work again

**Comment by someone5678** [comment]
YT for Android Q release version 15.28.35 has released! 
https://apkcombo.com/en-pl/apk-downloader/?device=&arch=&android=10&q=com.google.android.youtube

**Comment by th3an7** [comment]
> YT for Android Q release version 15.28.35 has released!
> https://apkcombo.com/en-pl/apk-downloader/?device=&arch=&android=10&q=com.google.android.youtube

Updated... YTM as well...
Shame my notifications for YT app updates are not working - would've update earlier 😄

**Comment by someone5678** [comment]
YT 15.29.35 is released!
https://apkcombo.com/en-pl/apk-downloader/?device=&arch=&android=10&q=com.google.android.youtube

**Comment by th3an7** [comment]
From now on I'll start commenting new commits with this issue number - that way I won't have to comment everytime I'll update hooks cause it automatically adds `added a commit that referenced this issue` with new YT version number 😄

**Comment by someone5678** [comment]
YT 15.30.36 released!
https://apkcombo.com/en-pl/apk-downloader/?device=&arch=&android=&q=com.google.android.youtube

**Comment by someone5678** [comment]
YT 15.31.36 is released!
https://apkcombo.com/youtube/com.google.android.youtube/

**Comment by hiishamz** [comment]
I don't know what happened.
But after the last update Youtube app kept force closing. I uninstalled it then re-installed it again and now everything is back to normal.
Thanks @th3an7 for your awesome work.

**Comment by th3an7** [comment]
> I don't know what happened.
> But after the last update Youtube app kept force closing. I uninstalled it then re-installed it again and now everything is back to normal.
> Thanks @th3an7 for your awesome work.

If it happens again, I recommend checking Xposed logs and/or logcat - should show why it's crashing...
Glad you've sorted that out 😄

**Comment by someone5678** [comment]
Youtube 15.32.34 is out!

https://apkcombo.com/en-pl/apk-downloader/?q=com.google.android.youtube

**Comment by HpNoTiQ56** [comment]
> # Updated
> ###### Also bonus - you can now check for classes updates yourselves - for more info check new branch on my fork [HERE](https://github.com/th3an7/YouTubeBackgroundPlayback/tree/IDA-Script)
> ###### (still needs hooks updates here but you can make your own fork, change links and build module .apk. That way you won't have to wait for my updates 😄 )

Nice work ! 
I'll try to create my repo next time!

**Comment by th3an7** [comment]
Small update - I've pushed new classes for 15.34.36 (commit https://github.com/pylerSM/YouTubeBackgroundPlayback/commit/bc50faca31f3631b8d456ffcada899e3f5a173db) - please let me know if it works (and/or if there are any errors in Xposed logs) as the first class has changed a little...

**Comment by thebestnom** [comment]
Working great!, Thanks as always

**Comment by Apologize327** [comment]
Hi, YTM 3.81.52 is not working, but 3.81.51 works well. 
Thanks @th3an7  for your work

**Comment by th3an7** [comment]
> Hi, YTM 3.81.52 is not working, but 3.81.51 works well.
> Thanks @th3an7 for your work

Latest supported YTM version is 3.79.50... Dunno how 3.81.51 is working despite no apk released (unless I'm blind) - latest one is 3.81.52... will update hooks for it in a moment...

Also for future - YTM has it's own reporting thread #362 - not an issue but I like to have them separated 😄

**Comment by Apologize327** [comment]
> > Hi, YTM 3.81.52 is not working, but 3.81.51 works well.
> > Thanks @th3an7 for your work
> 
> Latest supported YTM version is 3.79.50... Dunno how 3.81.51 is working ~despite no apk released (unless I'm blind)~ yup... I'm blind - latest one is 3.81.52... will update hooks for it in a moment...
> 
> Also for future - YTM has it's own reporting thread #362 - not an issue but I like to have them separated 😄
> 
> EDIT: Updated... they've changed something but I didn't add anything new just yet - hope the old method still works - let me know if it still doesn't work 🤞

OK, I'll report in thread [#362](https://github.com/pylerSM/YouTubeBackgroundPlayback/issues/362) next time.
I build apk with Android Studio by downloading latest source code :)  YTM 3.81.51 works fine indeed, but when it updated by Google Play, it's not working. Just now I uninstalled YTM and re-install it, and it's working, so werid.
Thanks @th3an7  a lot.

**Comment by th3an7** [comment]
> > > Hi, YTM 3.81.52 is not working, but 3.81.51 works well.
> > > Thanks @th3an7 for your work
> > 
> > 
> > Latest supported YTM version is 3.79.50... Dunno how 3.81.51 is working ~despite no apk released (unless I'm blind)~ yup... I'm blind - latest one is 3.81.52... will update hooks for it in a moment...
> > Also for future - YTM has it's own reporting thread #362 - not an issue but I like to have them separated 😄
> > EDIT: Updated... they've changed something but I didn't add anything new just yet - hope the old method still works - let me know if it still doesn't work 🤞
> 
> OK, I'll report in thread [#362](https://github.com/pylerSM/YouTubeBackgroundPlayback/issues/362) next time.
> I build apk with Android Studio by downloading latest source code :) YTM 3.81.51 works fine indeed, but when it updated by Google Play, it's not working. Just now I uninstalled YTM and re-installed it, and it's working, so werid.
> Thanks @th3an7 a lot.

Weird indeed... Also: you don't need to build your own .apk of module if you need it - you can check [my fork](https://github.com/th3an7/YouTubeBackgroundPlayback) (it's exact same as this one) but I've set up automatic builds with github actions (those are "debug" builds without signatures). There are also pre-built .apks hosted on my gdrive (links in relases tab on that fork) - those are the same that were in Xposed repo before removing (same signatures etc... all built by me)

**Comment by thebestnom** [comment]
Currently not having a computer with IDA, and I accidentally updated to 15.36.37, do you mind adding it?
Btw, if you have any way to use your script without IDA, it would be great

**Comment by th3an7** [comment]
> Currently not having a computer with IDA, and I accidentally updated to 15.36.37, do you mind adding it?
> Btw, if you have any way to use your script without IDA, it would be great

Sorry it took so long... 
Added 15.36.37 per request

As for script - I'll try to figure out something - if I find anything I'll update that branch 😅

**Comment by thebestnom** [comment]
Thank as always!!

Ill try testing installing IDA later this week so I could see how to that

**Comment by thebestnom** [comment]
oh no, IDA free doesn't support python... how fun, Ill try to port this to idc late,

**Comment by th3an7** [comment]
> oh no, IDA free doesn't support python... how fun, Ill try to port this to idc late,

I'll upload my .idc script in a second to that branch...
Also, if there are any questions about those - please make new issue on my fork... let's not spam this thread as it's lagging a bit as it is now 😆

**Comment by someone5678** [comment]
err I'm still not get used to using IDA so... I report YT new version release here

15.42.36 released

https://apkcombo.com/en-pl/apk-downloader/?q=com.google.android.youtube

**Comment by kozilord** [comment]
Downloaded YouTube from the Play Store, newest version seems to be 15.47.36

**Comment by th3an7** [comment]
> Downloaded YouTube from the Play Store, newest version seems to be 15.47.36

Added in https://github.com/pylerSM/YouTubeBackgroundPlayback/commit/cfa005b40643fb4cfc57ab6291d0bd55e9acce28
I'm stupid and I've missclicked enter while typing commit name so that's why the name is cut 🤦 

Anyway... should work again now

Also added support for YTM 4.04.53

**Comment by th3an7** [comment]
If anyone for some reason is still using this module instead of switching to Vanced, you probably need to get new module .apk
Thanks to @ohyeah521 it now supports function parameters (which means support for newer YT app versions)

You can either compile new module .apk yourself or if you want, you can get debug version that is automatically built with GitHub Actions [HERE](https://github.com/th3an7/YouTubeBackgroundPlayback/suites/1696280217/artifacts/31889278) (remember to unpack .zip and probably uninstall your version first... this one is not signed with any keys - still works like any normal, signed version)

If link above doesn't work, go [HERE](https://github.com/th3an7/YouTubeBackgroundPlayback/actions), choose most recent build and scroll down to "Artifacts" section

**Comment by preethamrn** [comment]
Does the hooks-3.json file need to be updated for each new version of YouTube? The latest apk build doesn't seem to be working on 16.16.38.

**Comment by th3an7** [comment]
> Does the hooks-3.json file need to be updated for each new version of YouTube? The latest apk build doesn't seem to be working on 16.16.38.

I've updated hooks for 16.16.38 - it should work again for that version (just force close YT app or reboot your device)

> Edit: Tried to create a pull request with the steps in the IDA-Script branch but it seems like without IDA Pro I can't get the classes (IDA Free doesn't seem to work). Could some help add the new version? Thanks

I'm not sure but .idc scripts should work in free version of IDA - python ones only work in PRO afaik... that's why I've ported .py to .idc
They are broken anyway (at least normal YT one) and need small fix - second class has changed significantly since I wrote them and it can't be found (1st and 3rd are working just fine) - I'll try to find new hex so it will work again


In the meantime I recommend checking @ohyeah521 branch [HERE](https://github.com/ohyeah521/YouTubeBackgroundPlayback) - he updates hooks regularly so the module should work just fine with them - you just need to recompile module .apk and [change links with hooks directing to his branch](https://github.com/pylerSM/YouTubeBackgroundPlayback/blob/master/java/com/pyler/youtubebackgroundplayback/YouTubeBackgroundPlayback.java#L47-L48)

**Comment by preethamrn** [comment]
Oh. I was only trying with YT (not YTM) so that might be it. I'm not familiar with IDA so I wasn't sure what I was looking for.

Thanks for the explanation as well as the update.
